src/storage/txn/actions/prewrite.rs

```diff
@@ -41,6 +41,28 @@ pub fn prewrite<S: Snapshot>(
     secondary_keys: &Option<Vec<Vec<u8>>>,
     pessimistic_action: PrewriteRequestPessimisticAction,
     expected_for_update_ts: Option<TimeStamp>,
+) -> Result<(TimeStamp, OldValue)> {
+    prewrite_with_generation(
```

Comment:
How about adding `generation` as an input parameter of `prewrite` function, if it's `None` related procesing could be skipped.

Reply:
That's my first thought too. But it exposes `generation` to many places where the users of this function don't have to know about it. By this wrapper we hide that information for most of the usages.
Besides, it saves a lot of modifications in those places, especially in tests.